File: Lucky Numbers[11-September-2023]/Solution.cpp
//   >> CPP CODE <<

class Solution{
public:
    bool check(int n,int i)
    {
        if(n<i) return true;
        if(n%i==0) return false;
        
        return check(n-n/i,i+1);
    }
    bool isLucky(int n) {
        return check(n,2);
    }
};

File: Lucky Numbers[11-September-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    def isLucky(self, n):
        def check(n,i):
            if(n<i): return True
            if(n%i==0): return False
            
            return check(n-n//i,i+1)
        
        return check(n,2)
